packages/imperative/src/config/cmd/convert-profiles/convert-profiles.handler.ts

```diff
@@ -0,0 +1,195 @@
+            for (const { name, type, error } of convertResult.profilesFailed) {
+                if (name != null) {
+                    params.response.console.error(`Failed to load ${type} profile "${name}":\n    ${error}`);
+                } else {
+                    params.response.console.error(`Failed to find default ${type} profile:\n    ${error}`);
+                }
+            }
+        }
+
+        params.response.console.log("");
+        const teamConfig = ImperativeConfig.instance.config;
+        teamConfig.api.layers.activate(false, true);
+        teamConfig.api.layers.merge(convertResult.config);
+        ConfigSchema.updateSchema();
+        await teamConfig.save(false);
+
+        const oldProfilesDir = `${profilesRootDir.replace(/[\\/]$/, "")}-old`;
+        try {
+            fs.renameSync(profilesRootDir, oldProfilesDir);
+        } catch (error) {
+            params.response.console.error(`Failed to rename profiles directory to ${oldProfilesDir}:\n    ${error}`);
+        }
+
+        const cliBin = ImperativeConfig.instance.rootCommandName;
+        // TODO Implement config edit command
+        params.response.console.log(`Your new profiles have been saved to ${teamConfig.layerActive().path}.\n` +
+            `Run "${cliBin} config edit --global-config" to open this file in your default editor.\n\n` +
+            `Your old profiles have been moved to ${oldProfilesDir}.\n` +
+            `Run "${cliBin} config convert-profiles --delete" if you want to completely remove them.`);
+    }
+
+    /**
+     * Retrieve list of obsolete CLI plug-ins that should be uninstalled.
+     * @returns List of plugins to uninstall containing the following properties:
+     *  - `name` - Name of plugin to uninstall
+     *  - `preuninstall` - Optional callback to be run before uninstall
+     *  - `postuninstall` - Optional callback to be run after uninstall
+     */
+    private getObsoletePlugins(): IPluginToUninstall[] {
+        const obsoletePlugins: IPluginToUninstall[] = [];
+
+        if (ImperativeConfig.instance.hostPackageName === this.ZOWE_CLI_PACKAGE_NAME) {
+            let credMgrSetting = AppSettings.instance.get("overrides", "CredentialManager");
+            if (credMgrSetting === ImperativeConfig.instance.hostPackageName) {
+                credMgrSetting = undefined;
+            }
+            obsoletePlugins.push({
+                name: typeof credMgrSetting === "string" ? credMgrSetting : this.ZOWE_CLI_SECURE_PLUGIN_NAME,
+                preUninstall: credMgrSetting ? this.disableCredentialManager : undefined
+            });
+        }
+
+        return obsoletePlugins;
+    }
+
+    /**
+     * Get the number of old profiles present in the CLI home dir.
+     * @param profilesRootDir Root profiles directory
+     * @returns Number of old profiles found
+     */
+    private getOldProfileCount(profilesRootDir: string): number {
```

Comment:
I'm guessing an application developer can recreate this function if needed, but would it be best to provide this a a utility ?

Reply:
Discussed in person and decided this is OK to leave private.